Dockerfile

```diff
@@ -1,41 +1,38 @@
 # Pelican Production Dockerfile
 
-FROM node:20-alpine AS yarn
-#FROM --platform=$TARGETOS/$TARGETARCH node:20-alpine AS yarn
-
-WORKDIR /build
-
-COPY . ./
-
-RUN yarn config set network-timeout 300000 \
-    && yarn install --frozen-lockfile \
-    && yarn run build:production
-
 FROM php:8.3-fpm-alpine
 # FROM --platform=$TARGETOS/$TARGETARCH php:8.3-fpm-alpine
 
 COPY --from=composer:latest /usr/bin/composer /usr/local/bin/composer
 
 WORKDIR /var/www/html
 
+RUN touch .env
+
 # Install dependencies
 RUN apk update && apk add --no-cache \
     libpng-dev libjpeg-turbo-dev freetype-dev libzip-dev icu-dev \
     zip unzip curl \
     caddy ca-certificates supervisor \
     && docker-php-ext-install bcmath gd intl zip opcache pcntl posix pdo_mysql
 
-# Copy the Caddyfile to the container
-COPY Caddyfile /etc/caddy/Caddyfile
+# Install dependencies with Composer
+COPY composer.json composer.lock ./
+COPY app/helpers.php ./app/
```

Comment:
This definitely confuses me even more. Definitely don't want to reference this file anywhere in Docker.

Reply:
These 3 files are the bare minimum required to run `composer install`. The reason `app/helpers.php` is required as well, is because it's specified as an autoload file in `composer.json`, so `composer install --optimize-autoloader` looks for it.

Reply:
Also the reason behind splitting the install and build up like this is just optimizing for build caching. By only copying the dependency lists first, those first layers only need to rerun when the dependencies change. Then the entire codebase is copied as late as possible to minimize needlessly rerunning. GitHub actions can also be configured to utilize caching (although I didn't do that in this PR), which, with these changes, should help speed up the builds a bit.

Although this doesn't have as big of an impact anymore because of the Vite frontend rewrite. The majority of the build time is now taken up by `yarn run build`, but regardless, I tried to move it down as far as possible.